eclair-core/src/main/scala/fr/acinq/eclair/blockchain/electrum/ElectrumClient.scala

```diff
@@ -16,64 +16,61 @@
 
 package fr.acinq.eclair.blockchain.electrum
 
-import java.io.InputStream
 import java.net.InetSocketAddress
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props, Stash, Terminated}
 import akka.io.{IO, Tcp}
 import akka.util.ByteString
 import fr.acinq.bitcoin._
-import fr.acinq.eclair.Globals
-import fr.acinq.eclair.blockchain.CurrentBlockCount
 import fr.acinq.eclair.blockchain.bitcoind.rpc.{Error, JsonRPCRequest, JsonRPCResponse}
 import org.json4s.JsonAST._
 import org.json4s.jackson.JsonMethods
 import org.json4s.{DefaultFormats, JInt, JLong, JString}
 import org.spongycastle.util.encoders.Hex
 
 import scala.annotation.tailrec
-import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.ExecutionContext
 import scala.concurrent.duration._
-import scala.util.Random
 
-class ElectrumClient(serverAddresses: Seq[InetSocketAddress]) extends Actor with Stash with ActorLogging {
+class ElectrumClient(serverAddress: InetSocketAddress)(implicit val ec: ExecutionContext) extends Actor with Stash with ActorLogging {
 
   import ElectrumClient._
   import context.system
 
   implicit val formats = DefaultFormats
 
   val newline = "\n"
-  val connectionFailures = collection.mutable.HashMap.empty[InetSocketAddress, Long]
   val socketOptions = Tcp.SO.KeepAlive(true) :: Nil
-
+  var addressSubscriptions = Map.empty[String, Set[ActorRef]]
+  var scriptHashSubscriptions = Map.empty[BinaryData, Set[ActorRef]]
+  val headerSubscriptions = collection.mutable.HashSet.empty[ActorRef]
   val version = ServerVersion("2.1.7", "1.1")
+  val statusListeners = collection.mutable.HashSet.empty[ActorRef]
+  val keepHeaders = 100
+
+  var reqId = 0L
+
+  self ! Tcp.Connect(serverAddress, options = socketOptions)
+
   // we need to regularly send a ping in order not to get disconnected
-  context.system.scheduler.schedule(30 seconds, 30 seconds, self, version)
+  val versionTrigger = context.system.scheduler.schedule(15 seconds, 15 seconds, self, version)
 
   override def unhandled(message: Any): Unit = {
     message match {
       case _: Tcp.ConnectionClosed =>
-        val nextAddress = nextPeer()
-        log.warning(s"connection failed, trying $nextAddress")
-        self ! Tcp.Connect(nextAddress, options = socketOptions)
+        log.info(s"connection to $serverAddress closed")
         statusListeners.map(_ ! ElectrumDisconnected)
-        context.system.eventStream.publish(ElectrumDisconnected)
-        context become disconnected
+        versionTrigger.cancel()
```

Comment:
Why cancell the task only in that case? Why not call `versionTrigger.cancel()` from e.g. `aroundPostStop`?